lib/Tumblr/StreamBuilder/FencepostRanking/FencepostRankedStream.php

```diff
@@ -729,6 +729,8 @@ public function log_commit_new_fence_stats(
                 $per_blog_count_string = Helpers::json_encode($uniq_blog_ids);
             } catch (\JsonException $e) {
                 $per_blog_count_string = '';
+                $log = StreamBuilder::getDependencyBag()->getLog();
+                $log->exception($e, $this->get_identity());
             }
             $log = StreamBuilder::getDependencyBag()->getLog();
```

Comment:
```suggestion
            $log = $log ?? StreamBuilder::getDependencyBag()->getLog();
```

since it could already exist now

Reply:
... or maybe we should just _always_ get this before the try/catch and avoid having to get it twice

Reply:
This doesn't matter since `StreamBuilder::getDependencyBag()` is a static field. There is no performance hit calling it twice. I would keep it as is since that's the simpler version